productivity-suite/app/legacy-app/src/components/pages/Login.tsx

```diff
@@ -16,10 +17,14 @@ export function Login() {
       return (
         getBus(ref.current).listen<{ username: string }>(
           "authentication",
-          (authDetails) => {
+          async (authDetails) => {
             if (authDetails?.username) {
-              auth.login(authDetails.username);
-              navigate(from, { replace: true });
+              await auth.login(authDetails.username);
+              const todoLists = await getTodoLists(authDetails.username);
+              getBus(ref.current!).dispatch("todo-list-selected", {
+                name: todoLists[todoLists.length - 1].name,
+              });
```

Comment:
This doesn't feel like the right solution here.
The Login page shouldn't know about the Todos page.
Can't we fix the Todos page so that it works this out for itself?

Reply:
Yeah I agree with you, if we don't want the login functionality to select the list for the user then I think we may need to refactor the todo lists selection logic we applied to fix the navigation issues

I am not sure if the current solution can handle it or if we may need to refactor back to the previous implementation and try to find a different solution for the navigation issues

what do you think?

Reply:
Let's talk on Monday about this. I have not looked into it but we should definitely not be in a situation where a view cannot initialize itself.

Reply:
I've applied the workaround we've discussed about, hopefully it's ok for now 🙂